src/main/java/org/terasology/climateConditions/alterationEffect/BodyTemperatureAlterationEffect.java

```diff
@@ -0,0 +1,124 @@
+     * the other applicable effect systems so that they can contribute with their own body temperature alteration effect
+     * related modifiers.
+     *
+     * @param instigator The entity who applied the body temperature alteration effect.
+     * @param entity The entity that the body temperature alteration effect is being applied on.
+     * @param id Stores information on type of body temperature change alteration - change depends on whether
+     *         temperature is decreasing or increasing.
+     * @param magnitude The magnitude of the body temperature alteration effect.
+     * @param duration The duration of the body temperature alteration effect.
+     */
+    @Override
+    public void applyEffect(EntityRef instigator, EntityRef entity, String id, float magnitude, long duration) {
+        // First, determine if the entity already has a swim speed component attached. If so, just replace the speed
+        // multiplier, and then save the component. Otherwise, create a new one and attach it to the entity.
+        AffectBodyTemperatureComponent affectBodyTemperature =
+                entity.getComponent(AffectBodyTemperatureComponent.class);
+        if (affectBodyTemperature == null) {
+            affectBodyTemperature = new AffectBodyTemperatureComponent();
+        }
+        affectBodyTemperature.postMultiplier = magnitude;
+        affectBodyTemperature.id = id;
+        entity.addOrSaveComponent(affectBodyTemperature);
+
+        // Send out this event to collect all the duration and magnitude modifiers and multipliers that can affect this
+        // effect.
+        OnEffectModifyEvent effectModifyEvent = entity.send(new OnEffectModifyEvent(instigator, entity, magnitude, duration, this, id));
+        long modifiedDuration = 0;
+        boolean modifiersFound = false;
+
+        // If the effect modify event is consumed, don't apply this effect.
+        if (!effectModifyEvent.isConsumed()) {
+            /*
+            Get the magnitude result value and the shortest duration, and assign them to the modifiedMagnitude and
+            modifiedDuration respectively.
+
+            The shortest duration is used as the effect modifier associated with that will expire in the shortest
+            amount of time, meaning that this effect's total magnitude and next remaining duration will have to be
+            recalculated.
+            */
+            float modifiedMagnitude = effectModifyEvent.getMagnitudeResultValue();
+            modifiedDuration = effectModifyEvent.getShortestDuration();
+
+            // If there's at least one duration and magnitude modifier, set the effect's magnitude and the
+            // modifiersFound flag.
+            if (!effectModifyEvent.getDurationModifiers().isEmpty() && !effectModifyEvent.getMagnitudeModifiers().isEmpty()) {
+                affectBodyTemperature.postMultiplier = modifiedMagnitude;
```

Comment:
This is the portion I was concerned about ...
We talked about having a constant addition the the change bodyTemperature by using postMultipliers in the AffectBodyTemperatureEvent when using the woolen vest..
The thing is, in all the alteration effects, the magnitude is treated as a Multiplier so having a postMultiplier in this effect might end up getting a different result than intended when OnEffectModifyEvent is used to get the resultant modified Magnitude